docs/custom_models.md

```diff
@@ -0,0 +1,104 @@
+
+## Model platform integrations
+
+### Hugging Face
+
+Example:
+
+```yaml
+model_deployments:
+  - name: vllm/pythia-70m
+    model_name: eleutherai/pythia-70m
+    tokenizer_name: EleutherAI/gpt-neox-20b
+    max_sequence_length: 2048
+    client_spec:
+      class_name: "helm.clients.huggingface_client.HuggingFaceClient"
+      args:
+        pretrained_model_name_or_path: EleutherAI/pythia-70m
+```
+
+Note: If `pretrained_model_name_or_path` is omitted, the model will be loaded from Hugging Face Hub using `model_name` (_not_ `name`) by default.
+
+Examples of common arguments:
+
+- Loading from local disk: `pretrained_model_name_or_path: /path/to/my/model`
+- Revision: `revision: my_revision`
+- Quantization: `load_in_8bit: true`
+- Model precision: `torch_dtype: torch.float16`
+
+Then run:
+
+```
+helm-run --run-entries mmlu:subject=anatomy,model=eleutherai/pythia-70m --suite my_suite --max-eval-instances 5
+```
+
+Note: This uses Hugging Face local inference. It will attempt to use GPU inference if available, and use CPU inference otherwise. It is only able to use the first GPU. Multi-GPU inference is not supported. Every model needed by `helm-run` will be loaded on the same GPU - if evaluating multiple models, it is prudent to evaluate each model with a separate `helm-run` invocation.
+
+### vLLM
+
+```yaml
+model_deployments:
+  - name: vllm/pythia-70m
+    model_name: eleutherai/pythia-70m
+    tokenizer_name: EleutherAI/gpt-neox-20b
+    max_sequence_length: 2048
+    client_spec:
+      class_name: "helm.clients.vllm_client.VLLMClient"
+      args:
+        base_url:  http://mymodelserver:8000/v1/
+```
+
+Set `base_url` to the URL of your inference server. On your inference server, run vLLM's OpenAI compatible server with:
+
+```
+python -m vllm.entrypoints.openai.api_server --model EleutherAI/pythia-70m
+```
+
+Then run:
+
+```
+helm-run --run-entries mmlu:subject=anatomy,model=eleutherai/pythia-70m --suite my_suite --max-eval-instances 5
+```
+
+### Together AI
+
+```yaml
+model_deployments:
+  - name: together/llama-2-7b
+    model_name: meta/llama-2-7b
+    tokenizer_name: meta-llama/Llama-2-7b-hf
+    max_sequence_length: 4094
+    client_spec:
+      class_name: "helm.clients.together_client.TogetherClient"
+      args:
+        together_model: togethercomputer/llama-2-7b
+```
+
+Then run:
+
+```
+helm-run --run-entries mmlu:subject=anatomy,model=meta/llama-2-7b --suite my_suite --max-eval-instances 5
+```
+
+Note: If `together_model` is omitted, the Together model with `model_name` (_not_ `name`) will be used by default.
+
+Note: This model may not be currently available on Together AI. Consult [Together AI's Inference Models documentation](https://docs.together.ai/docs/inference-models) for a list of currently available models and corresponding model strings.
```

Comment:
Add a note to remember to register Together credentials?

Reply:
Added.